lib/Document.php

```diff
@@ -191,14 +195,57 @@ public function download(?bool $postProcess = true, ?string $encoding = null): s
         return $contents;
     }
 
+    /**
+     * Downloads the document data as a stream.
+     * 
+     * @param resource|string|StreamInterface|null $output Optionnaly copy data stream to the given output.
+     *
+     * @param ?string &$encoding Get the document encoding from the HTTP request.
+     *
+     * @return StreamInterface The raw (unencrypted) document stream..
+     */
+    public function downloadStream($output = null, string &$encoding = null): StreamInterface {
+        try {
+            $response = $this->client->request('GET', $this->url, ['stream' => true]);
+        } catch (\GuzzleHttp\Exception\ClientException $e) {
+            $response = $e->getResponse();
+            if ($response->getStatusCode() == 404) {
+                throw new RuntimeException("Document Report not Found ({$response->getStatusCode()}): {$response->getBody()}");
+            } else {
+                throw $e;
+            }
+        }
+        
+        // trying to detect the document charset/encoding
+        $parsed = Header::parse($response->getHeader('content-type'));
+        foreach ($parsed as $header) {
+            if (isset($header['charset'])) {
+                $encoding = $header['charset'];
+                break;
+            }
+        }
+        $stream = $response->getBody();
+        if (\strtolower((string) $this->compressionAlgo) == 'gzip') {
```

Comment:
i think this should be accessible without the backslash -- builtin functions are in global namespace as far as i know

Reply:
It's an habit I've because some library like guzzle uses namespaced ovrriden functions.

I'll remove it :)